packages/react/src/hooks/useConnectorClient.ts

```diff
@@ -80,11 +80,22 @@ export function useConnectorClient<
   })
   const enabled = Boolean(status !== 'disconnected' && (query.enabled ?? true))
 
+  const addressRef = useRef<UseAccountReturnType<config>['address']>(address)
+
   // biome-ignore lint/nursery/useExhaustiveDependencies: `queryKey` not required
   useEffect(() => {
-    // invalidate when address changes
-    if (address) queryClient.invalidateQueries({ queryKey })
-    else queryClient.removeQueries({ queryKey }) // remove when account is disconnected
+    if (!address) {
```

Comment:
wouldn't this always removeQueries on page refresh? Not sure if there are edge cases like offline-first apps etc. Although perhaps re-initialising new walletClient on page refresh is a good thing

Reply:
Yup, good point & I think it is actually okay to revalidate on refresh

Reply:
Yeah I suppose you are right, client is not that critical to re-create on page load and forget the old one. As oppose to things like account, that is more important to get from persistent storage for offline-first apps

Reply:
idk if we should be removing queries on refresh if we don't need to

Reply:
ok yea, in this case @BrickheadJohnny perhaps you can check for `!!addressRef.current` ? and I would probably remove it from useRef initialisation, this feels like can lead to weird behaviour. I think it's better to have it null or undefined by default. wdyt?

Reply:
Actually I take it back